src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php

```diff
@@ -672,42 +719,7 @@ protected function typeTinyInteger(Fluent $column)
      */
     protected function typeSmallInteger(Fluent $column)
     {
-        return $this->generatableColumn('smallint', $column);
-    }
-
-    /**
-     * Create the column definition for a generatable column.
-     *
-     * @param  string  $type
-     * @param  \Illuminate\Support\Fluent  $column
-     * @return string
-     */
-    protected function generatableColumn($type, Fluent $column)
-    {
-        if (! $column->autoIncrement && is_null($column->generatedAs)) {
-            return $type;
-        }
-
-        if ($column->autoIncrement && is_null($column->generatedAs)) {
-            return with([
-                'integer' => 'serial',
-                'bigint' => 'bigserial',
-                'smallint' => 'smallserial',
-            ])[$type];
-        }
-
-        $options = '';
-
-        if (! is_bool($column->generatedAs) && ! empty($column->generatedAs)) {
-            $options = sprintf(' (%s)', $column->generatedAs);
-        }
-
-        return sprintf(
-            '%s generated %s as identity%s',
-            $type,
-            $column->always ? 'always' : 'by default',
-            $options
-        );
```

Comment:
This seems like a pretty huge change for Laravel? We are changing the entire way we define auto-incrementing IDs in Postgres? Is this necessary?

Reply:
It is not a change at all, it is necessary because PostgreSQL has different `alter column` syntax for column type and other column attributes so we have to separate type definitions and modifiers to handle that in a most clear way.

for example:
```
alter table `t1` alter column `col1` type bigserial. alter column `col1`, add generated always as identity, ....
```

Reply:
The `generatableColumn` method has changed to `modifyGeneratedAs` method.